src/Button/Button.svelte

```diff
@@ -1,92 +1,76 @@
 <script>
   /**
    * Specify the kind of button
-   * @type {"primary" | "secondary" | "tertiary" | "ghost" | "danger" | "danger-tertiary" | "danger-ghost"} [kind="primary"]
+   * @type {"primary" | "secondary" | "tertiary" | "ghost" | "danger" | "danger-tertiary" | "danger-ghost"}
    */
   export let kind = "primary";
 
   /**
    * Specify the size of button
-   * @type {"default" | "field" | "small"} [size="default"]
+   * @type {"default" | "field" | "small"}
    */
   export let size = "default";
 
-  /**
-   * Set to `true` for the icon-only variant
-   * @type {boolean} [hasIconOnly=false]
-   */
+  /** Set to `true` for the icon-only variant */
   export let hasIconOnly = false;
 
   /**
    * Specify the icon from `carbon-icons-svelte` to render
-   * @type {typeof import("carbon-icons-svelte/lib/Add16").default} [icon]
+   * @type {typeof import("carbon-icons-svelte/lib/Add16").default}
```

Comment:
Instead of typing the icon prop with the default `Add16` icon, wouldn’t it be cleaner using the `typeof CarbonIcon` class?—it applies to the rest of icon imports.

```ts
/**
 * Specify the icon from `carbon-icons-svelte` to render
 * @type {typeof import("carbon-icons-svelte").CarbonIcon}
 */
export let icon = undefined;
```

Although in this way you can’t restrict the icon size to e.g. `16`… 🤔

Reply:
Nice suggestion. The icon size should not matter.